cmd/kubeapps-apis/plugins/fluxv2/packages/v1alpha1/server.go

```diff
@@ -293,3 +331,82 @@ func getHelmIndexFileFromURL(indexURL string) (*helmrepo.IndexFile, error) {
 	index.SortEntries()
 	return &index, nil
 }
+
+// getClient ensures a client getter is available and uses it to return the client.
+func (s *Server) GetClient(ctx context.Context) (dynamic.Interface, error) {
+
+	if s.clientGetter == nil {
+		return nil, status.Errorf(codes.Internal, "server not configured with configGetter")
+	}
+	client, err := s.clientGetter(ctx)
+	if err != nil {
+		return nil, status.Errorf(codes.FailedPrecondition, fmt.Sprintf("unable to get client : %v", err))
+	}
+	return client, nil
+}
+
+// extractToken returns the token passed through the gRPC request in the "authorization" metadata
+// It is equivalent to the A"uthorization" usual HTTP 1 header
+// For instance: authorization="Bearer abc" will return "abc"
+func extractToken(ctx context.Context) (string, error) {
```

Comment:
Great - this is what I expected the crux of the initial PR to be - testing that we get the token correctly.

Reply:
So this functionality would be included as part of the `clientGetter` function passed to the plugin, so the plugin doesn't need to know or care how the auth is done.